### DISCUSSION TITLE
[2024-04-29] GUI Team Demo Day

### DISCUSSION THREAD

**Discussion by farmaazon** [enso-org/enso#9809] [Weekly Demo Day]
### @farmaazon 

* #9691 and #9755 are merged: the project should work after hibernating. Did not test if the dashboard reconnects correctly.
* #9771

[Screencast from 2024-04-23 12-57-01.webm](https://github.com/enso-org/enso/assets/3919101/873925db-1c9b-43e5-8144-6503c362baa0)

* #9775 - It's easier to click "add component button"

[Screencast from 2024-04-24 10-51-57.webm](https://github.com/enso-org/enso/assets/3919101/004aede5-ea00-4e30-9a3d-6e2574e3b755)

And the node expands/shrinks only when selection change ends (the selection box "drag" is released).

[Screencast from 2024-04-24 10-56-05.webm](https://github.com/enso-org/enso/assets/3919101/744e7579-80dd-4d44-a545-7af812239be2)

* https://github.com/enso-org/enso/pull/9788 - now you can click and drag edge from output port, instead of clicking twice

### @Frizi 

Paweł focused on bringing new CI tool. It required fixing some problems of the dependency tree of our JS packages.

### @vitvakatu 

* #9737 - the placeholders are displayed correctly, but there's a known issue of missing drop-downs in these arguments.

<img width="925" alt="Screenshot 2024-04-18 at 2 43 09 PM" src="https://github.com/enso-org/enso/assets/6566674/fbdce484-ac0b-4e30-8577-1c9dc419dffe">

**Comment by kazcw** [comment]
- #9734 (#9456)
- #9731
- #9781 (fixes #9703; implemented updating selection/viewport after creating nodes; fixed mouse placement and added multi-node mouse placement)
- #9784 (#9416)

**Comment by mwu-tow** [comment]
* There was an issue with `hades` server (self-hosted Windows runners). Due to some Docker corruption, the runners were down. Eventually I had to set the server from scratch again. As part of this, I've updated the setup instructions and applied some minor tweaks and updates to the environment.
* GitHub introduced new macOS 14 arm64 runners to the general availability. The `macos-latest` label now points to the new version. For now I've fixed the version macos to 12, so we retain the old behavior. We cannot simply switch to the 13/14 versions, as the `npm install` step is consistently causing the runners to get stuck.
* The installer and uninstaller are close to being finished. I'm cleaning the code, fixing the build issues that occur between the platforms (much of the code is Windows-only) and testing the behavior.

**Comment by hubertp** [comment]
Relevant backend changes:
 - `Table.join` work https://github.com/enso-org/enso/pull/9410
 - In-memory suggestions #9751 
 - Ongoing work on Y.js integration #9528 
 - AI integration #9679
